### PR TITLE
Angus/1694 virtualised region list

### LINKED ISSUE #1694: migrate region list to virtualised table
The region list is currently a simple HTML table. This is fine for very small region sets, but @duidae has made major improvements to the render speed for regions (see #1664). The bottleneck is now back on the region viewer. 

We should make this a simple virtualised table. We could use BlueprintJS tables, but I think we want to try to migrate away from them in future. Should we try just using a simple react virtualized table library, such as [react-window](https://github.com/bvaughn/react-window)?

### PR DESCRIPTION
**Pull request by veggiesaurus**
Closes #1694 and #1252 by using a `react-window` to render a virtualised region list.

Thanks to @acdo2002 for some help with the CSS changes.

### CODE REVIEW

**Review by acdo2002 — APPROVED** [review]
Thanks Angus solved the critical bug, the css works fine on Ubuntu & MacOS with Chrome & Firefox browsers!

File: src/components/RegionList/RegionListComponent.tsx [hunk 1]
```diff
@@ -196,104 +266,94 @@ export class RegionListComponent extends React.Component<WidgetProps> {
...
-                            <th style={{width: RegionListComponent.CENTER_COLUMN_DEFAULT_WIDTH}}>{frame.validWcs ? "Center" : "Pixel Center"}</th>
-                            {showSizeColumn && <th style={{width: RegionListComponent.SIZE_COLUMN_DEFAULT_WIDTH}}>{frame.validWcs ? "Size" : "Size (px)"}</th>}
-                            {showRotationColumn && <th style={{width: RegionListComponent.ROTATION_COLUMN_DEFAULT_WIDTH}}>P.A. (deg)</th>}
-                        </tr>
-                    </thead>
-                    <tbody className={appStore.darkTheme ? "dark-theme" : ""}>{rows}</tbody>
-                </HTMLTable>
-                <ReactResizeDetector handleWidth handleHeight onResize={this.onResize}></ReactResizeDetector>
+                <div className={classNames("region-list-table", {"bp3-dark": darkTheme})}>
+                    <FixedSizeList itemSize={25} height={25} itemCount={1} width="100%" className="list-header">
```
Comment (YuHsuan-Hwang) [inline src/components/RegionList/RegionListComponent.tsx]: we could define a static variable `HEADER_ROW_HEIGHT` for 25

File: src/components/RegionList/RegionListComponent.tsx [hunk 2]
```diff
@@ -196,104 +266,94 @@ export class RegionListComponent extends React.Component<WidgetProps> {
...
+                        {headerRenderer}
+                    </FixedSizeList>
+                    <FixedSizeList onItemsRendered={this.onListRendered} height={tableHeight - 35} itemCount={this.validRegions.length} itemSize={RegionListComponent.ROW_HEIGHT} width="100%">
```
Comment (YuHsuan-Hwang) [inline src/components/RegionList/RegionListComponent.tsx]: maybe changing `tableHeight - 35` to `tableHeight - HEADER_ROW_HEIGHT - padding * 2` is more clear

**Review by YuHsuan-Hwang — COMMENTED** [review]
The changes look nice. I have only very minor comments

**Review by kswang1029 — APPROVED** [review]
this works and looks good 👍 thanks @veggiesaurus @acdo2002

**Review by YuHsuan-Hwang — APPROVED** [review]

**Pull request merged**